src/freenas/usr/local/bin/truenas-grub.py

```diff
@@ -16,12 +17,12 @@ def get_serial_ports():
     advanced = query_config_table("system_advanced", prefix="adv_")
     kernel_extra_options = advanced.get("kernel_extra_options") or ""
 
-    # check for /data/.vendor
+    vendor = Vendors.TRUENAS_SCALE
     try:
         with open("/data/.vendor", "r") as f:
-            vendor = json.loads(f.read()).get("name", "TrueNAS Scale")
-    except FileNotFoundError:
-        vendor = "TrueNAS Scale"
+            vendor = json.loads(f.read()).get("name", Vendors.TRUENAS_SCALE)
```

Comment:
Should we have a check that the name read from the vendor file matches an value in our enum?

Reply:
At a minimum we should probably log an error message.

Reply:
This will crash (by design) on proper TrueNAS machines so don't want to pollute logs unnecessarily. However, we could narrow the exception scope.
```
try:
    with open(
except FileNotFoundError:
    pass
except Exception:
    log.here

Reply:
This will crash (by design) on proper TrueNAS machines so don't want to pollute logs unnecessarily. However, we could narrow the exception scope.
```
try:
    with open(
except FileNotFoundError:
    pass
except Exception:
    log.here

Reply:
This will crash (by design) on proper TrueNAS machines so don't want to pollute logs unnecessarily. However, we could narrow the exception scope.
```
try:
    with open(
except FileNotFoundError:
    pass
except Exception:
    log.here

Reply:
> Should we have a check that the name read from the vendor file matches an value in our enum?

I don't think we need to do this, as an invalid vendor will result in default TrueNAS functionality (just with a different Grub entry name). A user would have to really go out of their way to add a vendor during the install process.



> This will crash (by design) on proper TrueNAS machines so don't want to pollute logs unnecessarily. However, we could narrow the exception scope.

Gotcha, will do!